src/main.rs

```diff
@@ -221,12 +295,53 @@ where
     writeln!(file, "source=(\"{}\")", source)?;
     writeln!(file, "sha256sums=(\"{}\")", sha256)?;
     writeln!(file)?;
+    // Include the prepare, build and check steps for non-binary package.
+    if no_bin {
+        writeln!(file, "prepare() {{")?;
+        writeln!(file, "    cd $pkgname-$pkgver")?;
+        writeln!(file, "    export RUSTUP_TOOLCHAIN=stable")?;
+        writeln!(
+            file,
+            "    cargo fetch --locked --target \"$(rustc -vV | sed -n 's/host: //p')\""
+        )?;
+        writeln!(file, "}}")?;
+        writeln!(file)?;
+        writeln!(file, "build() {{")?;
+        writeln!(file, "    cd $pkgname-$pkgver")?;
+        writeln!(file, "    export RUSTUP_TOOLCHAIN=stable")?;
+        writeln!(file, "    export CARGO_TARGET_DIR=target")?;
+        writeln!(file, "    cargo build --frozen --release --all-features")?;
+        writeln!(file, "}}")?;
+        writeln!(file)?;
+        writeln!(file, "check() {{")?;
+        writeln!(file, "    cd $pkgname-$pkgver")?;
+        writeln!(file, "    export RUSTUP_TOOLCHAIN=stable")?;
+        writeln!(file, "    cargo test --frozen --all-features")?;
```

Comment:
For tests I agree that `--all-features` is probably what they want.

Note that for the builder, this will cause all dependencies to be built a second time, but if this is irritating they can build with `--nocheck` as supplied by `makepkg` and other tools.